packages/contract/src/test-utils/launch-test-node.test.ts

```diff
@@ -0,0 +1,236 @@
+}
+
+/**
+ * @group node
+ */
+describe('launchTestNode', () => {
+  test('kills the node after going out of scope', async () => {
+    let url = '';
+
+    {
+      using launched = await launchTestNode();
+
+      const { provider } = launched;
+
+      url = provider.url;
+      await provider.getBlockNumber();
+    }
+
+    await waitUntilUnreachable(url);
+
+    const { error } = await safeExec(async () => {
+      const p = await Provider.create(url);
+      await p.getBlockNumber();
+    });
+
+    expect(error).toMatchObject({
+      message: 'fetch failed',
+    });
+  });
+
+  test('kills the node if error happens post-launch on contract deployment', async () => {
+    const spy = vi.spyOn(setupTestProviderAndWalletsMod, 'setupTestProviderAndWallets');
+
+    const { error } = await safeExec(() =>
+      launchTestNode({ deployContracts: ['invalid location'] })
+    );
+    expect(error).toBeDefined();
+    // Verify that error isn't due to
+    expect(spy).toHaveBeenCalled();
+
+    const {
+      provider: { url },
+    } = spy.mock.results[0].value as { provider: { url: string } };
+
+    // test will timeout if the node isn't killed
+    await waitUntilUnreachable(url);
+  });
+
+  test('a contract can be deployed', async () => {
+    using launched = await launchTestNode({
+      deployContracts: [{ contractDir: pathToContractRootDir }],
+    });
+
+    const {
+      contracts: [contract],
+    } = launched;
+
+    const gasPrice = contract.provider.getGasConfig().minGasPrice;
+
+    const response = await contract.functions
+      .test_function()
+      .txParams({
+        gasPrice,
+        gasLimit: 10_000,
+      })
+      .call();
+    expect(response.value).toBe(true);
+  });
+
+  test('a contract can be deployed by providing just the path', async () => {
+    // #region deploy-contract
+    using launched = await launchTestNode({
+      deployContracts: [pathToContractRootDir],
+    });
+
+    const {
+      contracts: [contract],
+      provider,
+      wallets,
+    } = launched;
+
+    const response = await contract.functions.test_function().call();
+    // #endregion deploy-contract
+    expect(response.value).toBe(true);
+    expect(provider).toBeDefined();
+    expect(wallets).toBeDefined();
+  });
+
+  test('multiple contracts can be deployed with different wallets', async () => {
+    // #region multiple-contracts-and-wallets
+    using launched = await launchTestNode({
+      walletConfig: {
+        count: 2,
+        assets: [AssetId.A, AssetId.B],
+        coinsPerAsset: 2,
+        amountPerCoin: 1_000,
+      },
+      deployContracts: [
+        pathToContractRootDir,
+        { contractDir: pathToContractRootDir, walletIndex: 1, options: { storageSlots: [] } },
+      ],
+    });
+
+    const {
+      contracts: [contract1, contract2],
+      wallets: [wallet1, wallet2],
+    } = launched;
+    // #endregion multiple-contracts-and-wallets
```

Comment:
@nedsalk we need to place code snippets within `app/doc-snippets`

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1356/commits/4752fe77d7a8a42275c78f9173896f497178b07b